pkg/db/executor.go

```diff
@@ -12,40 +12,50 @@ import (
 )
 
 func NewExecutor(config cfg.Config, logger log.Logger, name string, backoffType string, notifier ...exec.Notify) exec.Executor {
+	return NewExecutorWithOptions(config, logger, name, backoffType, notifier)
+}
+
+func NewExecutorWithOptions(config cfg.Config, logger log.Logger, name string, backoffType string, notifier []exec.Notify, options ...exec.BackoffExecutorOption) exec.Executor {
```

Comment:
Why is a function to create some kind of executor getting backoff executor options? it should just be executor options passed here